super_orchestrator/src/api_docker/misc_structs.rs

```diff
@@ -144,26 +144,26 @@ pub(crate) fn port_bindings_to_bollard_args(
     Option<HashMap<String, HashMap<(), ()>>>,
     Option<HashMap<String, Option<Vec<bollard::secret::PortBinding>>>>,
 ) {
-    Some(
-        pbs.iter()
-            .map(|pb| {
-                let port_proto = format!("{}/{}", pb.container_port, pb.protocol);
-                (
-                    (port_proto.clone(), HashMap::new()),
-                    (
-                        port_proto.clone(),
-                        Some(vec![bollard::secret::PortBinding {
-                            host_port: pb
-                                .host_port
-                                .or(Some(pb.container_port))
-                                .as_ref()
-                                .map(ToString::to_string),
-                            host_ip: pb.host_ip.as_ref().map(ToString::to_string),
-                        }]),
-                    ),
-                )
-            })
-            .unzip(),
-    )
-    .unzip()
+    let mut port_map =
+        HashMap::<String, Option<Vec<bollard::secret::PortBinding>>>::with_capacity(pbs.len());
+    let mut open_ports = HashMap::<String, HashMap<(), ()>>::with_capacity(pbs.len());
```

Comment:
Is that a `HashMap<(), ()>`? Can you replace this with a `HashSet<String>` or a `HashMap<String, ()>` (In some cases the hashset API is not good)